bootstrap.sh

```diff
@@ -16,6 +16,16 @@ INSTALL_DIR=$ROOT/install
 
 cd deps
 
+lwgrp=lwgrp-1.0.2
+dtcmp=dtcmp-1.0.3
+
+if [ ! -f ${lwgrp}.tar.gz ] ; then
+  wget https://github.com/hpc/lwgrp/releases/download/v1.0.2/${lwgrp}.tar.gz
+fi
+if [ ! -f ${dtcmp}.tar.gz ] ; then
+  wget https://github.com/hpc/dtcmp/releases/download/v1.0.3/${dtcmp}.tar.gz
+fi
+
```

Comment:
Rather than download the tarballs, you clone the repos and checkout the version instead?  That follows convention with the other dependencies.  I actually prefer to clone the repos in case I need to make a quick edit to one of the dependencies during testing; you can easily see what you've changed in git.

Reply:
Not in this case.  These are autotools builds, and so the tarballs includes the needed configure script.

Reply:
To elaborate a bit more, one could in theory clone the lwgrp and dtcmp repos and then build the configure script.  However, I don't want our end users to deal with that.  Autotools is notoriously bad with compatabilty between versions.  Unless our end users have the right autotools versions installed, they can't create the configure script.  So then they have to download and build autotools to create the configure script to then finally build the libraries themselves.  The makedist tarballs here are far more portable for end users.

But we could offer a bootstrap for developers and another bootstrap for end users.  While at it, our bootstrap for end users should probably build with optimization by default and debug as an option.

Reply:
Should we convert dtcmp and lwgrp to CMake? Wouldnt be too difficult.

Reply:
@adammoody I've actually found the opposite to be true.  In fact, we had to build the latest zfs source tarballs on Fedora 32, because the ones we were building on RHEL 7 generated a configure script that was too old, causing build errors (https://github.com/openzfs/zfs/issues/8345).  

Note that lwgrp does include a configure script in the git repo.

Reply:
Yes, converting to cmake would be a clean way to handle this.